rust/cyclotron-node/src/lib.rs

```diff
@@ -0,0 +1,485 @@
+use chrono::{DateTime, Utc};
+use cyclotron_core::{
+    base_ops::{JobInit, JobState, WaitingOn},
+    manager::{ManagerConfig, QueueManager},
+    worker::Worker,
+    PoolConfig,
+};
+
+use neon::{
+    handle::Handle,
+    prelude::{Context, FunctionContext, ModuleContext},
+    result::{JsResult, NeonResult},
+    types::{JsNull, JsNumber, JsPromise, JsString, JsValue},
+};
+use once_cell::sync::OnceCell;
+use serde::de::DeserializeOwned;
+use serde_json::Value;
+use tokio::runtime::Runtime;
+use uuid::Uuid;
+
+static WORKER: OnceCell<Worker> = OnceCell::new();
+static MANAGER: OnceCell<QueueManager> = OnceCell::new();
+static RUNTIME: OnceCell<Runtime> = OnceCell::new();
+
+fn runtime<'a, C: Context<'a>>(cx: &mut C) -> NeonResult<&'static Runtime> {
+    RUNTIME
+        .get_or_try_init(Runtime::new)
+        .or_else(|e| cx.throw_error(format!("failed to create tokio runtime: {}", e)))
+}
+
+// The general interface for calling our functions takes a JSON serialized stirng,
+// because neon has no nice serde support for function arguments (and generally.
+// rippping objects from the v8 runtime piece by piece is slower than just passing
+// a since chunk of bytes). These are convenience functions for converting between
```

Comment:
```suggestion
// a single chunk of bytes). These are convenience functions for converting between
```

(This kind of trails off, too.)